### PR TITLE
VMSS and KV control spec update

### PR DESCRIPTION
**Pull request by asimks**

### CODE REVIEW

File: Control coverage/Feature/KeyVault.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
+| <b>ControlId:</b><br>Azure_KeyVault_Audit_Enable_Diagnostics_Log<br><br><b>DisplayName:</b><br>Diagnostics logs must be enabled with a retention period of at least 365 days.<br><br><b>Description: </b><br> Diagnostics logs must be enabled with a retention period of at least 365 days. |<b> ARM API to fetch diagnostic settings for specified resource: </b> <br> /{resourceId}/providers/microsoft.insights<br>/diagnosticSettings?api-version=2017-05-01-preview <br><br><b>Properties:</b><br> properties/logs/enabled<br> properties/logs/retentionPolicy/enabled<br> properties/logs/retentionPolicy/days| <b>Scope: </b> All KeyVault in a subscription<br><br><b>Config: </b><br>DiagnosticRetentionPeriodForever:"0"<br>DiagnosticRetentionPeriodMin:"365"<br><br><b>Passed: </b><br>Diagnostic setting are correctly configured<br>i.e.Diagnostic setting is enabled <br>and retaining log for atleast 365 days.<br><br><b>Failed: </b><br>Diagnostic setting is diabled <br>or not retaining log for 365 days. |
```
Comment (gvaradarajan-msft) [inline Control coverage/Feature/KeyVault.md:5]: Typo - `diabled` to be replaced with `disabled`.

File: Control coverage/Feature/KeyVault.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+| ControlId | Dependent Azure API(s) and Properties | Control Spec |
```
Comment (gvaradarajan-msft) [inline Control coverage/Feature/KeyVault.md:1]: A general comment on all updates - It is usually a "Spec-let" that is included here and not the actual "Spec". While the additional information helps, you may want to include just the "Spec-let" to be uniform with what is currently available.

File: Control coverage/Feature/VirtualMachineScaleSet.md [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
...roviders<br>/Microsoft.Compute/virtualMachineScaleSets<br>?api-version=2019-07-01 <br><br><b>Properties:</b><br>properties/storageProfile/osDisk/osType<br>properties/virtualMachineProfile/extensionProfile/extension<br><br><b> ARM API to get all the VM instances for the specified VMSS: </b> <br> /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers<br>/Microsoft.Compute/virtualMachineScaleSets/{resourceName}/virtualMachines<br>?api-version=2019-07-01 <br><br><b>Properties:</b><br> instance/resources(Microsoft.Compute/virtualMachines/extensions)| <b>Scope: </b> All Windows VMSS in a subscription<br><br><b>Config: </b><br>ExtensionType:"IaaSAntimalware"<br>Publisher:"Microsoft.Azure.Security"<br><br><b>Passed: </b><br>Antimalware Malware extension is deployed at VMSS model<br> and all it's VM instances with Auto Upgrade<br> to minor version enabled and Realtime protection enabled.<br><br><b>Failed: </b><br>AntiMalware extension not deplyed at VMSS model<br> or at one or more VM instances<br> or AntiMalware extension is present but<br> Auto Upgarde to minor version is disabled <br>or AntiMalware extension is present but<br> Auto Realtime protection is disabled. |
```
Comment (gvaradarajan-msft) [inline Control coverage/Feature/VirtualMachineScaleSet.md:10]: Typo - `deplyed` to be replaced with `deployed`.

Comment (gvaradarajan-msft) [inline Control coverage/Feature/VirtualMachineScaleSet.md:10]: Typo - `Upgarde` to be replaced with `Upgrade`.

File: Control coverage/Feature/VirtualMachineScaleSet.md [hunk 4]
```diff
@@ -1,6 +1,16 @@
 ## VirtualMachineScaleSet
```
Comment (gvaradarajan-msft) [inline Control coverage/Feature/VirtualMachineScaleSet.md:2]: Please retain `Spec-let` over `Spec`.

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]

File: Control coverage/Feature/VirtualMachineScaleSet.md [hunk 5]
```diff
@@ -1,6 +1,16 @@
...
+| <b>ControlId:</b><br>Azure_VirtualMachineScaleSet_SI_Missing_OS_Patches<br><br><b>DisplayName:</b><br>System updates on virtual machine scale sets must be installed.<br><br><b>Description: </b><br> Virtual Machine Scale Set must have all the required OS patches installed. |<b> ARM API to fetch all the security assessment for the specified subscription: </b> <br> /subscriptions/{subscriptionId}/providers<br>/Microsoft.Security/assessments<br>?api-version=2020-01-01 <br><br><b>AssessmentName:</b><br>bd20bd91-aaf1-7f14-b6e4-866de2f43146| <b>Scope: </b> All VMSS in a subscription<br><br><b>Config: </b>NA<br><br><b>Passed: </b><br>ASC assessment is in healthy state.<br><br><b>Failed: </b><br>ASC assessment is in unhealthy state. |
```
Comment (gvaradarajan-msft) [inline Control coverage/Feature/VirtualMachineScaleSet.md:13]: You may want to call out what fields are considered for the evaluation. For example, the status code from the assessment.

File: Control coverage/Feature/VirtualMachineScaleSet.md [hunk 6]
```diff
@@ -1,6 +1,16 @@
...
+| <b>ControlId:</b><br>Azure_VirtualMachineScaleSet_SI_Remediate_Security_Vulnerabilities<br><br><b>DisplayName:</b><br>Vulnerabilities in security configuration on your virtual machine scale sets must be remediated.<br><br><b>Description: </b><br> Vulnerabilities in security configuration on your virtual machine scale sets must be remediated. |<b> ARM API to fetch all the security assessment for the specified subscription: </b> <br> /subscriptions/{subscriptionId}/providers<br>/Microsoft.Security/assessments<br>?api-version=2020-01-01 <br><br><b>AssessmentName:</b><br>8941d121-f740-35f6-952c-6561d2b38d36| <b>Scope: </b> All VMSS in a subscription<br><br><b>Config: </b>NA<br><br><b>Passed: </b><br>ASC assessment is in healthy state.<br><br><b>Failed: </b><br>ASC assessment is in unhealthy state. |
```
Comment (gvaradarajan-msft) [inline Control coverage/Feature/VirtualMachineScaleSet.md:14]: You may want to call out what fields are considered for the evaluation. For example, the status code from the assessment.

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]

File: Control coverage/Feature/VirtualMachineScaleSet.md (hunk 6 above)
Comment (asimks) [reply Control coverage/Feature/VirtualMachineScaleSet.md:14]: Done

**Review by gvaradarajan-msft — APPROVED** [review]

**Pull request merged**